### PR TITLE
Script for publishing rustdocs at gh-pages

### PR DESCRIPTION
**Pull request by jimmychu0807**
This is a followup PR of #9856.

Per @TriplEight comment, now it is using a shell script to handle rustdocs release.

Usages:

```
# Examples:
rustdocs-release.sh -h                        # Shows help text
rustdocs-release.sh deploy monthly-2021-10    # Builds and deploys `monthly-2021-10` branch/tag rustdocs
rustdocs-release.sh deploy -l monthly-2021-10 # Builds and deploys `monthly-2021-10` branch/tag rustdocs, and `latest` symlink is pointing to the doc dir
rustdocs-release.sh remove monthly-2021-10    # Remove the rustdocs of `monthly-2021-10`
```

@sacha-l @NukeManDan 

rustdocs of `monthly-2021-12` and `monthly-2022-01` tags have been deployed using this script at:
- https://github.com/paritytech/substrate/commit/db4bd2038ff0d96b6a976296f2a16b60e7c84bb7, and 
- https://github.com/paritytech/substrate/commit/08302ebf08ea46d22039f8267c27df50b5f62ba7 respectively.

### CODE REVIEW

**Review by alvicsam — APPROVED** [review]
LGTM, I have only one comment: I think it would be better to publish `@jimmychu0807/index-tpl-crud` in `@parity` or `@substrate` at npmjs.com

**Review by radupopa2010 — APPROVED** [review]

**Comment by jimmychu0807** [comment]
Let me check the requirement of publishing a package under @substrate.

**Comment by jimmychu0807** [comment]
@alvicsam 
I have transferred [`index-tpl-crud` project](https://github.com/substrate-developer-hub/index-tpl-crud) to Substrate devhub. Will update the node published package to under @substrate namespace in my next PR. Would love to merge this first and have the team using it.

File: .maintain/rustdocs-release.sh [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+#   # Remove the rustdocs of `monthly-2021-10` from `gh-pages`.
+#   rustdocs-release.sh remove monthly-2021-10
+#
+# Dependencies:
+#   - @jimmychu0807/index-tpl-crud - https://www.npmjs.com/package/@jimmychu0807/index-tpl-crud
+#
+
+# Script setting
+# The git repo http URL
+REMOTE_REPO="https://github.com/paritytech/substrate.git"
```
Comment (TriplEight) [inline .maintain/rustdocs-release.sh:32]: is it used somewhere? If it is externally, then please `export`

**Review by TriplEight — APPROVED** [review]

File: .maintain/rustdocs-release.sh (hunk 1 above)
Comment (jimmychu0807) [reply .maintain/rustdocs-release.sh:32]: No, just some kind of constant set at the beginning of the script.

**Pull request merged**